onnxruntime/core/session/IOBinding.h

```diff
@@ -79,6 +79,12 @@ class IOBinding {
     */
   AllocatorPtr GetCPUAllocator(int id, onnxruntime::ProviderType provider_type) const;
 
+  /**
+    * clear inputs or outputs. IOBinding is stateful. There are cases we need to reset its state.
+    */
+  void ClearOutputs();
+  void ClearInputs();
+
  private:
```

Comment:
Do we need to reset the state or could we create a new instance of IOBinding instead?

Reply:
from PyTorch frontend, the trainer binds inputs and outputs between ort and pytorch tensors. When doing gradient accumulation, the trainer needs to clear outputs binding and to rebind outputs. This is to reset an all_fp32_gradients_finite output flag otherwise the next train_step will trigger all_reduce incorrectly. 
In this scenario, only ClearOutputs() is called. Inputs binding stays unchanged.

Reply:
My understanding is that it is more convenient from the Python frontend to be able to clear and reconstruct the outputs alone. ClearInputs() was added for symmetry.